From 397a9e7dc35a7b360bb8466996ae7755574dc04c Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 22 May 2017 17:36:16 -0700
Subject: [PATCH 2/2] lldb: Add -lxml2 to linker cmdline of xml is found

When cross compiling for systems where static libs
for libxml are not available cmake's detection mechanism
resort to linking with libxml.so but doesnt use -lxml2
liblldbHost.a however requires libxml on linker
cmdline _after_ itself so its use of symbols from libxml2
can be resolved. Here check for libxml2 being detected and
add it if its found.

Fixes

| ../../../../lib/liblldbHost.a(XML.cpp.o): In function `lldb_private::XMLDocument::Clear()':
| /usr/src/debug/lldb/5.0.0+gitAUTOINC+69edad7913_08d6b47db9_cf6c5b3386-r0/git/tools/lldb/source/Host/common/XML.cpp:29: undefined reference to `xmlFreeDoc'
| /usr/src/debug/lldb/5.0.0+gitAUTOINC+69edad7913_08d6b47db9_cf6c5b3386-r0/git/tools/lldb/source/Host/common/XML.cpp:29: undefined reference to `xmlFreeDoc'

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 source/Host/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source/Host/CMakeLists.txt b/source/Host/CMakeLists.txt
index 2b6f0e48a..1a910a9af 100644
--- a/source/Host/CMakeLists.txt
+++ b/source/Host/CMakeLists.txt
@@ -175,6 +175,10 @@ if (NOT LLDB_DISABLE_LIBEDIT)
   list(APPEND EXTRA_LIBS edit)
 endif()
 
+if (LIBXML2_FOUND)
+ list(APPEND EXTRA_LIBS xml2)
+endif ()
+
 add_lldb_library(lldbHost
   ${HOST_SOURCES}
   
-- 
2.16.1

